core/src/main/java/org/elasticsearch/gateway/ReplicaShardAllocator.java

```diff
@@ -186,6 +187,7 @@ public boolean allocateUnassigned(RoutingAllocation allocation, long allocateUna
                      * see {@link org.elasticsearch.cluster.routing.RoutingService#clusterChanged(ClusterChangedEvent)}).
                      */
                     changed = true;
+                    allocation.setHasPendingAsyncFetch();
```

Comment:
I'm not sure this one is correct? If we got here it means we know everything about the current nodes, but couldn't find a  (good) copy. Strictly speaking, we don't have an inflight async fetch. Also , I think it's OK to rebalance here?